### DISCUSSION TITLE
Implement the modified cam-clay model for unsaturated soil behavior

### DISCUSSION THREAD

**Discussion by Auubinno** [thelfer/tfel#633] [Q&A] [answered]
Hello, I need to implement the modified Cam-Clay model for unsaturated soils in Code_Aster, and I am considering using MFront as it seems much simpler than modifying the source code.

In the implementation, I intend to include a term that considers the increment of saturation due solely to the increment of stress. Therefore, the saturation must be updated during the numerical integration step along with the stresses. Is it possible, through MFront, to modify the internal variable related to the change in saturation present in the HM model?

**Comment by thelfer** [comment]
@Auubinno You shall receive information from the `aster` team soon.

**Comment by SRaude** [comment, marked as answer]
@Auubinno,

I think you can find some information by following these two links:
- [Barcelone](https://gitlab.com/codeaster/src/-/blob/main/mfront/Barcelone.mfront?ref_type=heads)
- [MCC](https://gitlab.com/codeaster/src/-/blob/main/mfront/MCC.mfront?ref_type=heads)

These constitutive models have recently been implemented in the unstable version of code_aster.

Simon

**Reply by Auubinno** [reply]
Thanks @thelfer and @SRaude.

I'll take a look at those files.
